src/citrine/_serialization/properties.py

```diff
@@ -340,8 +360,17 @@ def _serialize(self, value: list) -> list:
             serialized.append(self.element_type.serialize(element))
         return serialized
 
+    def _set_elements(self, value):
+        elems = []
+        for sub_val in value:
+            if issubclass(type(sub_val), self.element_type.underlying_types):
+                elems.append(sub_val)
+            else:
+                elems.append(self.element_type.deserialize(sub_val))
```

Comment:
This is actually the line I'm more worried about.  I don't think `typing.Type[Property]` has a `deserialize` method.